### PR TITLE
Experiment add graphQL

### PR DESCRIPTION
**Pull request by InfAurora**
# PR Description
Initial installation steps for graphQL with apollo server; Following this [tutorial](https://www.youtube.com/watch?v=XzE-PzALyDc&t=1258s).

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Additional Context or Relevant Issues

Please include additional context.

## Issue Link

Please link the original issue URL.

### CODE REVIEW

**Comment by AhmedHussein-1** [comment]
Nice GraphQL!

**Review by LuisGonzalezJr — COMMENTED** [review]
we should move the schema and resolvers

**Review by jtooleyful — DISMISSED** [review]

File: .eslintrc.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
         "plugin:cypress/recommended",
         "plugin:prettier/recommended"
     ],
-    "ignorePatterns": ["cypress.config.ts"],
+    "ignorePatterns": ["cypress.config.ts", "graphql.ts"],
```
Comment (mamohamed252) [inline .eslintrc.json:15]: remember to uncomment graphql.ts later

**Review by LuisGonzalezJr — COMMENTED** [review]
looks like apollo server is having issues running this but works fine in postman

**Comment by mamohamed252** [comment]
> looks like apollo server is having issues running this but works fine in postman

and resolve npm install for build

File: src/api.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+export const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      refetchOnMount: false,
+      refetchOnWindowFocus: false,
+      refetchOnReconnect: false,
+    },
+  },
+});
```
Comment (mamohamed252) [inline src/api.ts]: add new line for end of file

**Review by mamohamed252 — DISMISSED** [review]

**Review by LuisGonzalezJr — APPROVED** [review]

**Review by AhmedHussein-1 — APPROVED** [review]

**Review by jtooleyful — APPROVED** [review]

File: .github/workflows/pr.yml [hunk 3]
```diff
@@ -29,5 +29,5 @@ jobs:
       - name: npm test
         run: |
           echo ----RUNNING TESTS-----
-          npm run test-once
-          echo ----RUNNING TESTS----
+          npm run dev & npm run test-once
```
Comment (mamohamed252) [inline .github/workflows/pr.yml:32]: single & allows dev to run along side run test while double & has to wait for dev to complete which it does not.

**Review by mamohamed252 — APPROVED** [review]

**Pull request merged**